Sources/Sentry/SentrySession.m

```diff
@@ -9,60 +9,107 @@ @implementation SentrySession
 
 @synthesize flagInit = _init;
 
-- (instancetype)initWithReleaseName:(NSString *)releaseName
+/**
+ * Default private constructor. We don't name it init to avoid the overlap with the default init of
+ * NSObject, which is not available as we specified in the header with SENTRY_NO_INIT.
+ */
+- (instancetype)initDefault
 {
     if (self = [super init]) {
         _sessionId = [NSUUID UUID];
         _started = [SentryCurrentDate date];
         _status = kSentrySessionStatusOk;
         _sequence = 1;
         _errors = 0;
-        _init = @YES;
         _distinctId = [SentryInstallation id];
+    }
+
+    return self;
+}
+
+- (instancetype)initWithReleaseName:(NSString *)releaseName
+{
+    if (self = [self initDefault]) {
+        _init = @YES;
         _releaseName = releaseName;
     }
     return self;
 }
 
 - (instancetype)initWithJSONObject:(NSDictionary *)jsonObject
 {
-    // Note this doesn't use the main init method since it should only init
-    // fields that exist in the JSON.
-    if (self = [super init]) {
-        _sessionId = [[NSUUID UUID] initWithUUIDString:[jsonObject valueForKey:@"sid"]];
-        _distinctId = [jsonObject valueForKey:@"did"];
-        NSString *startedString = [jsonObject valueForKey:@"started"];
-        if (nil != startedString) {
-            _started = [NSDate sentry_fromIso8601String:startedString];
-        }
-        NSString *timestampString = [jsonObject valueForKey:@"timestamp"];
-        if (nil != timestampString) {
-            _timestamp = [NSDate sentry_fromIso8601String:timestampString];
-        }
-        NSString *status = [jsonObject valueForKey:@"status"];
-        if ([@"ok" isEqualToString:status]) {
-            _status = kSentrySessionStatusOk;
-        } else if ([@"exited" isEqualToString:status]) {
-            _status = kSentrySessionStatusExited;
-        } else if ([@"crashed" isEqualToString:status]) {
-            _status = kSentrySessionStatusCrashed;
-        } else if ([@"abnormal" isEqualToString:status]) {
-            _status = kSentrySessionStatusAbnormal;
-        }
-        _sequence = [[jsonObject valueForKey:@"seq"] unsignedIntegerValue];
-        _errors = [[jsonObject valueForKey:@"errors"] unsignedIntegerValue];
+    // We use the default constructor here to set the non nullable values to a default values,
+    // because this could cause crashes, for example, in serialize.
+    // With this approach we avoid crashes and accept the tradeoff that some session data might not
+    // be 100% accurate.
+    if (self = [self initDefault]) {
+        id sid = [jsonObject valueForKey:@"sid"];
+        if (nil != sid && [sid isKindOfClass:[NSString class]]) {
```

Comment:
L: question, on Java/Kotlin, if you do an `instanceof` and the condition returns `true`, it's obviously not null, so the 1st check would not be necessary, that said, can we avoid `nil != sid` here?

Reply:
the same would apply for the other properties, if possible, eg `status`.

if this is not possible, we should consider adding the NPE check to the other conditions, eg `started`

Reply:
That's also true for Objective-C. I'm going to fix that. Thanks for the input :)

Reply:
already merged, but here `nil != sid` could be avoided as well

Reply:
@marandaneto see: https://github.com/getsentry/sentry-cocoa/pull/878